test/sim/abilities/cursedbody.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const assert = require('./../../assert');
+const common = require('./../../common');
+
+let battle;
+
+describe(`Cursed Body`, function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it.skip(`should be able to trigger on Z-moves`, function () {
```

Comment:
I actually wrote this test wrong, and your comment prompted me (thanks!) - Cursed Body can Disable the Z-move itself, **not** the base of the Z-move. Disable can't Disable the Z-move, but Cursed Body can, because yeah.

I am not familiar offhand with every mechanic listed but I believe all are accurate (save the Cursed Body one that I have corrected now); I will follow up after further testing on a couple of them.

Reply:
So, if Cursed Body can disable the Z-move itself, what about a move called by Dancer?

Reply:
Great follow-up! There was one thing mistaken in this list - Spite, Eerie Spell, and G-Max Depletion all can work on Max Moves, and they reveal the base move when doing so, like Grudge.

Reply:
> So, if Cursed Body can disable the Z-move itself, what about a move called by Dancer?

Yep!